### DISCUSSION TITLE
How to change ToolJet language?

### DISCUSSION THREAD

**Discussion by danddinh** [ToolJet/ToolJet#6375] [Q&A] [answered]
There's a guide here to create languages for UI: https://docs.tooljet.com/docs/contributing-guide/l10n/

However, the UI of ToolJet including the admin UI and app UI has no where to change language, a dropdown to select language or that kind of stuff. 

Anyone, how to change UI language pls?

**Comment by withshubh** [comment]
Hey @dandeancook 👋 

The language setting in the `.env` file can only be modified by an administrator. The file contains a language key, which is initially set to "en". Users have the option to enter a valid abbreviation for the language, which our system supports.

**Reply by danddinh** [reply]
> a valid abbreviation for the language

Is that option on ToolJet built-in UI? or used during app making?

**Reply by withshubh** [reply, marked as answer]
Please go through this guide: https://docs.tooljet.com/docs/contributing-guide/l10n

**Comment by liudonghua123** [comment]
> Hey @dandeancook 👋
> 
> The language setting in the `.env` file can only be modified by an administrator. The file contains a language key, which is initially set to "en". Users have the option to enter a valid abbreviation for the language, which our system supports.

From the code search https://github.com/ToolJet/ToolJet/blob/68f68a0351cad22b0ab3cec76bff2e0b54ef098c/frontend/src/index.jsx#L20, https://github.com/search?q=repo%3AToolJet%2FToolJet%20LanguageSelection%20&type=code.

I tried to config `LANGUAGE` in .env, but it did not work, it seems the `LanguageSelection` component is not used now.

**Comment by liudonghua123** [comment]
It's working when I tried to recreate the docker containers. But the translation seems not complete. I have followed the guidance on https://docs.tooljet.com/docs/contributing-guide/l10n/.

My translation is [here](https://github.com/ToolJet/ToolJet/pull/8353).

![image](https://github.com/ToolJet/ToolJet/assets/2276718/607e8c95-4b00-464f-96e8-a48919505834)

![image](https://github.com/ToolJet/ToolJet/assets/2276718/6a888173-99eb-4dcf-9583-7477dca80c2a)

**Reply by betterzjy** [reply]
thanks, I an also looking for Chinese translation
